### ISSUE TITLE
Failed Twitter Sign-in, 500 error

### ISSUE THREAD

**Issue by stumash** [igrigorik/vimgolf#249]
Clicking the 'Sign in with Twitter' button GETs

https://vimgolf.com/auth/twitter?x_auth_access_type=read&use_authorize=true

which gives a 500 error response.

In other words, I can't sign in with twitter. I get a 'We're sorry, but...' message.

**Comment by Hettomei** [comment]
here is the log from server :  (I can't help much currently)

```
018-06-15T14:51:02.558571+00:00 app[web.1]: Started GET "/auth/twitter?x_auth_access_type=read&use_authorize=true" for 172.xxx.xxx.xxx at 2018-06-15 14:51:02 +0000
2018-06-15T14:51:02.559303+00:00 app[web.1]: I, [2018-06-15T14:51:02.559228 #9]  INFO -- omniauth: (twitter) Request phase initiated.
2018-06-15T14:51:02.696192+00:00 app[web.1]:   
2018-06-15T14:51:02.696242+00:00 app[web.1]: OAuth::Unauthorized (403 Forbidden):
2018-06-15T14:51:02.696261+00:00 app[web.1]:   
2018-06-15T14:51:02.696310+00:00 app[web.1]: vendor/bundle/ruby/2.4.0/gems/oauth-0.5.4/lib/oauth/consumer.rb:236:in `token_request'
2018-06-15T14:51:02.696312+00:00 app[web.1]: vendor/bundle/ruby/2.4.0/gems/oauth-0.5.4/lib/oauth/consumer.rb:155:in `get_request_token'
2018-06-15T14:51:02.696313+00:00 app[web.1]: vendor/bundle/ruby/2.4.0/gems/omniauth-oauth-1.1.0/lib/omniauth/strategies/oauth.rb:28:in `request_phase'
2018-06-15T14:51:02.696314+00:00 app[web.1]: vendor/bundle/ruby/2.4.0/gems/omniauth-twitter-1.4.0/lib/omniauth/strategies/twitter.rb:61:in `request_phase'
2018-06-15T14:51:02.696315+00:00 app[web.1]: vendor/bundle/ruby/2.4.0/gems/omniauth-1.8.1/lib/omniauth/strategy.rb:224:in `request_call'
```

**Comment by dennisplosceanu** [comment]
Maybe there's an answer here: https://stackoverflow.com/questions/50826742/omniauth-twitter-with-rails-5-stopped-working-oauthunauthorized-403-forbidden ?

**Comment by stumash** [comment]
Is anyone else unable to sign in with twitter or is it just me?

**Comment by Hettomei** [comment]
Me too. There is a real problem

**Comment by dennisplosceanu** [comment]
It must be related to this:
https://twittercommunity.com/t/action-required-sign-in-with-twitter-users-must-whitelist-callback-urls/105342

There was 1 month of notice, it started a couple of days ago.

**Comment by igrigorik** [comment]
I believe this should be fixed now — thanks for the tip on twitter change. I had to whitelist the callback URL in the "app" on twitter.com. Please reopen if this is still an issue.

**Comment by dennisplosceanu** [comment]
Still the same issue.

**Comment by igrigorik** [comment]
o_O ... can't reproduce it here. Others?

**Comment by dennisplosceanu** [comment]
Maybe you registered the callback from a wrong build?

**Comment by Hettomei** [comment]
I can't register too

**Comment by igrigorik** [comment]
@dennisplosceanu I'm testing on production site. What do you mean by "wrong build"?

Not sure what's going on here.

**Comment by dennisplosceanu** [comment]
Yeah, what I wanted to ask was if you were testing on production.

**Comment by stumash** [comment]
still not working

**Comment by igrigorik** [comment]
Yes, testing on production site. 

q: have you been able to authenticate successfully before, or are you trying to create new accounts?
q: I'm assuming you're accessing http://www.vimgolf.com, correct?

**Comment by dennisplosceanu** [comment]
Ok, wow. I was trying on https://vimgolf.com/ .
It works with http.

**Comment by stumash** [comment]
@igrigorik , I was not accessing [http://www.vimgolf.com](http://www.vimgolf.com), but actually just [https://vimgolf.com](https://vimgolf.com) and [http://vimgolf.com](http://vimgolf.com).  Are those other two domains wrong?

**Comment by stumash** [comment]
in any case, the issue seems to be resolved so long as I visit [http://www.vimgolf.com](http://www.vimgolf.com) instead of the url without the `www` (w/ or w/o https).  I would love to know why the non-www version of the site is different but I guess it doesn't really matter.

TL;DR : Resolved 👍

**Comment by igrigorik** [comment]
Ah, figures.

- I've setup rules to redirect non-www to www for HTTP and HTTPS
- I've whitelisted HTTPS callback URLs — you should be able to login on HTTPS.

Created a separate bug for "migrate to HTTPS": https://github.com/igrigorik/vimgolf/issues/250

Closing, and hopefully this time it sticks. :)
